internal/app/orchestrator/orchestrator.go

```diff
@@ -50,6 +54,13 @@ const (
 // more details.
 type Orchestrator interface {
 	gcp.Cache
+
+	// shutdown takes an aggregated key and shuts down the go routine watching
+	// for upstream responses.
+	//
+	// This is currently used by tests to clean up channels, but can also be
+	// used by the main shutdown handler.
+	shutdown(string)
```

Comment:
Can we add a name for the parameter? I am guessing the string value is the key?

Reply:
Changed this to take a context and shutdown all upstream streams.